client/src/Backend/GameLogic/ContractsAPI.ts

```diff
@@ -1155,6 +1165,76 @@ export class ContractsAPI extends EventEmitter {
     return player;
   }
 
+  // PlayerProfile Functions
+
+  public async getAllPlayerProfiles(
+    onProgress?: (fractionCompleted: number) => void,
+  ): Promise<Map<string, PlayerProfile>> {
+    const nPlayers: number = (await this.makeCall<EthersBN>(this.contract.getNPlayers)).toNumber();
+
+    const profiles = await aggregateBulkGetter<{playerAddress:string, displayName:string}>(
+      nPlayers,
+      200,
+      async (start:number, end:number) => (await this.makeCall<{playerAddress:string, displayName:string}[]>(this.contract.bulkGetDisplayNames, [start, end])),
+      onProgress
+    );
+
+    const playerProfileMap: Map<EthAddress, PlayerProfile> = new Map();
+
+    for (const profile of profiles) {
+      playerProfileMap.set(
+        address(profile.playerAddress),
+        {
+          playerAddress: address(profile.playerAddress),
+          displayName: profile.displayName,
+        },
+      );
+    }
+
+    return playerProfileMap;
+  }
+
+  /**
+   * Get the display name for a plyaer by their address.
+   * @param playerAddress - The address of the player.
+   * @returns The display name of the player.
+   */
+  public async getDisplayName(playerAddress: EthAddress): Promise<string|undefined> {
+    try {
+      const displayName = await this.makeCall<string>(this.contract.getDisplayName, [playerAddress]);
+    } catch (error) {
+      console.error('Error fetching display name:', error);
+      return undefined;
+    }
+  }
+
+  /**
+   * Set the display name for a player.
+   * Only that player or the admin can call this method.
+   * @param playerAddress - The address of the player.
+   * @param newDisplayName - The new display name to set.
+   * @returns A transaction object for setting the display name.
+   */
+  public async setDisplayName(
```

Comment:
ContractsAPI only calls the contracts to get the data, not handle with the logic of submit transactions.

It is better to directly implement this function in GameManager